### PR TITLE
Showcase Page

### PR DESCRIPTION
**Pull request by LamboCreeper**
Requires #44

### CODE REVIEW

**Comment by LamboCreeper** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codesupport/website-frontend/pullRequest?prid=5205005&bid=17071883)

File: src/services/ShowcaseService.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        try {
+            const {data} = await axios.get(this.api);
+
+            if (data.status === "OK") {
+                return data.response;
+            } else {
+                return [];
+            }
+        } catch (error) {
+            throw new Error(`There was an error getting all the showcase posts.`);
```
Comment (LamboCreeper) [inline src/services/ShowcaseService.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/codesupport/website-frontend/pullRequest?prid=5205005)

**Comment by LamboCreeper** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codesupport/website-frontend/pullRequest?prid=5205005&bid=17071883)

File: src/services/ShowcaseService.js (hunk 1 above)
Comment (LamboCreeper) [inline src/services/ShowcaseService.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/codesupport/website-frontend/pullRequest?prid=5205005)

File: src/services/ShowcaseService.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                    contributors: post.contributorList.contributors.map((contributor) => ({
+                       id: contributor.user.id,
+                       alias: contributor.user.alias
+                    }))
+                }));
+            } else {
+                return [];
+            }
+        } catch (error) {
+            throw new Error(`There was an error getting all the showcase posts.`);
```
Comment (LamboCreeper) [inline src/services/ShowcaseService.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/codesupport/website-frontend/pullRequest?prid=5205005)

**Comment by LamboCreeper** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codesupport/website-frontend/pullRequest?prid=5205005&bid=17071883)

File: src/components/pages/Community.js [hunk 3]
```diff
@@ -1,7 +1,71 @@
...
+            });
+        }
+    }
+
+    componentDidMount() {
+        this.getShowcasePosts();
+    }
+
+    render() {
+        const {error, loaded, showcase_posts: showcasePosts } = this.state;
```
Comment (Jason2605) [inline src/components/pages/Community.js]: Very, very minor point so will approve anyways - I'm not sure i understand the rationale of the switch between snake_case and camelCase - could this not just be defined as camelCase when the state is initialised?

**Review by Jason2605 — APPROVED** [review]

File: src/components/pages/Community.js (hunk 3 above)
Comment (LamboCreeper) [reply src/components/pages/Community.js]: Valid point, I have this habit of making everything in a JSON object snake case. Will switch it over to camel case as it's pointless here.

**Pull request merged**